ts/components/ChooserListSearchComponent.tsx

```diff
@@ -0,0 +1,173 @@
+import I18n from "i18n-js";
+import * as pot from "italia-ts-commons/lib/pot";
+import { Text, View } from "native-base";
+import React from "react";
+import {
+  Image,
+  ImageSourcePropType,
+  ListRenderItem,
+  StyleSheet
+} from "react-native";
+import customVariables from "../theme/variables";
+import ChooserList from "./ChooserList";
+
+type OwnProps<T> = {
+  listState: ReadonlyArray<T>;
+  searchText: string;
+  keyExtractor: (item: T, index: number) => string;
+  renderItem: ListRenderItem<T>;
+  onSearchItemContainsText?: (item: T, searchText: string) => boolean;
+  noSearchResultsSourceIcon?: ImageSourcePropType;
+  noSearchResultsSubtitle?: string;
+};
+
+type Props<T> = OwnProps<T>;
+
+type State<T> = {
+  potFilteredListStates: pot.Pot<ReadonlyArray<T>, string>;
+};
+
+const styles = StyleSheet.create({
+  emptyContentWrapper: {
+    flex: 1,
+    alignItems: "center"
+  },
+  emptyTitle: {
+    fontSize: customVariables.fontSizeBase,
+    paddingRight: customVariables.contentPadding,
+    paddingLeft: customVariables.contentPadding,
+    paddingTop: customVariables.contentPadding,
+    textAlign: "center"
+  },
+  emptySubtitle: {
+    fontSize: customVariables.fontSizeBase,
+    paddingRight: customVariables.contentPadding,
+    paddingLeft: customVariables.contentPadding,
+    paddingBottom: customVariables.contentPadding,
+    textAlign: "center"
+  }
+});
+/**
+ * A component that renders a list of object that match a search text.
+ */
+class ChooserListSearchComponent<T> extends React.PureComponent<
+  Props<T>,
+  State<T>
+> {
+  constructor(props: Props<T>) {
+    super(props);
+    this.state = {
+      potFilteredListStates: pot.noneLoading
+    };
+  }
+
+  /**
+   * Filter only the object that match the searchText.
+   */
+  private generateListStateMatchingSearchTextAsync(
+    listState: ReadonlyArray<T>,
+    searchText: string
+  ): Promise<ReadonlyArray<T>> {
+    const { onSearchItemContainsText } = this.props;
+    return new Promise(resolve => {
+      const result = listState.filter(
```

Comment:
The actual code returns the whole list when the search text is empty (**""**), this is right.
But I guess we can avoid performing a filter when the searchText is empty so this special case should return the listState directly

Reply:
Done. In commit d405cb0